pkg/syncer/admissionhandler/admissionhandler.go

```diff
@@ -170,6 +172,47 @@ func StartWebhookServer(ctx context.Context, enableWebhookClientCertVerification
 			go containerOrchestratorUtility.HandleLateEnablementOfCapability(ctx, cnstypes.CnsClusterFlavorGuest,
 				common.LinkedCloneSupport, gcConfig.GC.Port, gcConfig.GC.Endpoint)
 		}
+		pvcsiConfigPath := cnsconfig.GetConfigPath(ctx)
+		watcher, err := fsnotify.NewWatcher()
+		if err != nil {
+			log.Errorf("failed to create fsnotify watcher. err=%v", err)
+			return err
+		}
+		go func() {
+			for {
+				log.Debugf("Waiting for event on fsnotify watcher")
+				select {
+				case event, ok := <-watcher.Events:
+					if !ok {
+						return
+					}
+					log.Debugf("fsnotify event: %q", event.String())
+					if event.Op&fsnotify.Remove == fsnotify.Remove {
```

Comment:
Can you add some comments?  Why restart the webhook when there is a "Remove" event? Do we need to watch for "Create" and "Write" events as well?

Reply:
Do we need to add back the watch if it is removed?  https://github.com/kubernetes-csi/external-snapshotter/blob/v6.1.0/pkg/validation-webhook/certwatcher.go#L142